python/ray/serve/scripts.py

```diff
@@ -397,11 +403,40 @@ def build(import_path: str, app_dir: str, output_path: Optional[str]):
             if property in deployment:
                 del deployment[property]
 
-    if output_path is not None:
-        if not output_path.endswith(".yaml"):
-            raise ValueError("FILE_PATH must end with '.yaml'.")
+    config_str = yaml.dump(
+        config, Dumper=ServeBuildDumper, default_flow_style=False, sort_keys=False
+    )
+    config_str += f"# Built using Ray v{ray.__version__}\n"
 
+    if output_path is not None:
         with open(output_path, "w") as f:
-            yaml.safe_dump(config, stream=f, default_flow_style=False, sort_keys=False)
+            f.write(config_str)
     else:
-        print(yaml.safe_dump(config, default_flow_style=False, sort_keys=False), end="")
+        print(config_str, end="")
```

Comment:
you can unify this with default f to sys.stdout

Reply:
Got it, I refactored it to:

```python
with open(output_path, "w") if output_path else sys.stdout as f:
    f.write(config_str)
```